File: main/gl_samples/data/scripts/scene03.py
import core
import math


class Scene03(core.System):

    def begin(self):
        self.t = 0.0

    def update(self, dt):
        self.t += dt

    def draw(self, draw_command: core.DrawCommand):
        draw_command.bind()
        draw_command.set_float("value", (math.sin(self.t)+1)/2)
        draw_command.draw()

    def end(self):
        print("End")

File: main/gl_samples/data/scripts/scene03.lua
Scene03 = System:new()

function Scene03:begin ()
    self.t = 0.0
end

function Scene03:update(dt)
    self.t += dt
end

function Scene03:draw(draw_command)
    draw_command.bind()
    draw_command.set_float("value", (math.sin(self.t)+1.0)/2.0)
    draw_command.draw()
end

function Scene03:end()
end